dbms/src/Functions/LeastGreatest.h

```diff
@@ -120,5 +122,484 @@ class FunctionVectorizedLeastGreatest : public IFunction
 private:
     const Context & context;
 };
+template <bool least, bool use_collator = true>
+struct LeastGreatestStringImpl
+{
+    static void mergeImpl(
+        const TiDB::TiDBCollatorPtr & collator,
+        size_t a_size,
+        size_t b_size,
+        const unsigned char * a_data,
+        const unsigned char * b_data,
+        ColumnString::Chars_t & c_data,
+        ColumnString::Offsets & c_offsets,
+        size_t i)
+    {
+        int res = 0;
+        auto pre_offset = StringUtil::offsetAt(c_offsets, i);
+        if constexpr (use_collator)
+            res = collator->compare(reinterpret_cast<const char *>(&a_data[0]), a_size, reinterpret_cast<const char *>(&b_data[0]), b_size);
+        else
+            res = memcmp(&a_data[0], &b_data[0], std::min(a_size, b_size));
+
+        if constexpr (least)
+        {
+            if (res < 0)
+            {
+                memcpy(&c_data[pre_offset], &a_data[0], a_size);
+                c_offsets[i] = pre_offset + a_size + 1;
+            }
+            else if (res == 0)
+            {
+                size_t size = std::min(a_size, b_size);
+                memcpy(&c_data[pre_offset], &b_data[0], size);
```

Comment:
`mem_utils::avx2_inline_memcpy` may be helpful.

Reply:
Actually, we have replace it globally, check `libs/libmemcpy/memcpy.cpp`

Reply:
> avx2_inline_memcpy

Nice, I will use inline_memcpy